public/app/plugins/datasource/cloudwatch/tracking.test.ts

```diff
@@ -0,0 +1,767 @@
+                    InstanceId: '*',
+                  },
+                  expression: '',
+                  hide: false,
+                  id: 'a',
+                  matchExact: true,
+                  metricEditorMode: 0,
+                  metricName: 'CPUUtilization',
+                  metricQueryType: 0,
+                  namespace: 'AWS/EC2',
+                  period: '',
+                  queryMode: 'Metrics',
+                  refId: 'B',
+                  region: 'default',
+                  sqlExpression: '',
+                  statistic: 'Average',
+                },
+                {
+                  alias: '',
+                  datasource: {
+                    type: 'cloudwatch',
+                    uid: 'abc',
+                  },
+                  dimensions: {
+                    InstanceId: 'i-123',
+                  },
+                  expression: '',
+                  id: 'a',
+                  matchExact: true,
+                  metricEditorMode: 0,
+                  metricName: 'CPUUtilization',
+                  metricQueryType: 0,
+                  namespace: 'AWS/EC2',
+                  period: '',
+                  queryMode: 'Metrics',
+                  refId: 'A',
+                  region: 'default',
+                  sqlExpression: '',
+                  statistic: 'Average',
+                },
+                {
+                  alias: '',
+                  datasource: {
+                    type: 'cloudwatch',
+                    uid: 'abc',
+                  },
+                  dimensions: {
+                    InstanceId: '*',
+                  },
+                  expression: 'a / ',
+                  hide: false,
+                  id: '',
+                  matchExact: true,
+                  metricEditorMode: 1,
+                  metricName: 'CPUUtilization',
+                  metricQueryType: 0,
+                  namespace: 'AWS/EC2',
+                  period: '',
+                  queryMode: 'Metrics',
+                  refId: 'B',
+                  region: 'default',
+                  sqlExpression: '',
+                  statistic: 'Average',
+                },
+              ] as CloudWatchQuery[],
+            },
+          })
+        );
+      }),
+    }),
+  };
+});
+
+describe('onDashboardLoadedHandler', () => {
+  it('should report a `grafana_ds_cloudwatch_dashboard_loaded` interaction ', () => {
+    expect(reportInteraction).toHaveBeenCalledWith('grafana_ds_cloudwatch_dashboard_loaded', {
+      dashboard_id: 'dashboard123',
+      grafana_version: 'v9.0.0',
+      org_id: 1,
+      user_id: 2,
+      logs_queries_count: 1,
+      metrics_queries_count: 21,
+      metrics_query_builder_count: 3,
+      metrics_query_code_count: 4,
+      metrics_query_count: 7,
+      metrics_search_builder_count: 9,
+      metrics_search_code_count: 5,
+      metrics_search_count: 14,
+      metrics_search_match_exact_count: 9,
```

Comment:
I am wondering if it will be difficult to understand which queries correspond to which event from looking at this assertion.
Would splitting up each query and showing which counters it increments be helpful to a future reader?

Reply:
hmm I think in general, I would agree that it's a good practice to have explicit tests and assertions like that. but at the same time, it would require one new dashboard per prop to test, so I think it may be overkill for this simple tracking logic? WDYT?